src/renderer/gl_engine/tvgGlTessellator.cpp

```diff
@@ -889,10 +889,17 @@ void Tessellator::tessellate(const Array<const RenderShape *> &shapes)
     this->fillRule = FillRule::NonZero;
 
     for (uint32_t i = 0; i < shapes.count; i++) {
-        auto cmds = shapes[i]->path.cmds.data;
-        auto cmdCnt = shapes[i]->path.cmds.count;
-        auto pts = shapes[i]->path.pts.data;
-        auto ptsCnt = shapes[i]->path.pts.count;
+        const RenderPath* path = nullptr;
+        RenderPath trimmedPath;
+        if (shapes[i]->trimpath()) {
+            if (!shapes[i]->stroke->trim.trim(shapes[i]->path, trimmedPath)) continue;
+            path = &trimmedPath;
+        } else path = &shapes[i]->path;
+
+        auto cmds = path->cmds.data;
+        auto cmdCnt = path->cmds.count;
+        auto pts = path->pts.data;
+        auto ptsCnt = path->pts.count;
 
         this->visitShape(cmds, cmdCnt, pts, ptsCnt);
```

Comment:
Let's go simpler:
`
this->visitShape(path->cmds.data, path->cmds.count, path->pts.data, path->pts.count);
`